dashboard/app/models/levels/bubble_choice.rb

```diff
@@ -162,6 +185,10 @@ def self.parent_levels(level_name)
     where("properties -> '$.sublevels' LIKE ?", "%\"#{level_name}\"%")
   end
 
+  def set_unless_nil(hash, key, value)
+    hash[key] = value unless value.nil?
+  end
```

Comment:
if we do decide we want a utility method like this, we should find a better place to put it; model files should only contain model-specific code, not utility code that we incidentally use within the model-specific logic

Reply:
It wasn't immediately clear to me what set_unless_nil did without taking a quick peek at the implementation.  (It wasn't clear that the function took a hash and the order of the params is not obvious.)

It looks like there's some additional repeated code when calling `I18n.t`.  I wonder if it would be more readable to make an I18n-specific helper (put somewhere with I18n code) that would allow us to write something like `translate_string(summary, :display_name)` or `translate_string_in_hash()` with optional named parameters as needed?